ios/RNSScreenStack.mm

```diff
@@ -165,9 +162,8 @@ - (void)navigationController:(UINavigationController *)navigationController
   // TODO: Improve this merge when StackHeaderConfig is merged
 #ifdef RN_FABRIC_ENABLED
   if ([view isKindOfClass:RNSScreenView.class]) {
-    RNSScreenStackHeaderConfigComponentView *config =
-        (RNSScreenStackHeaderConfigComponentView *)((RNSScreenView *)view).config;
-    [RNSScreenStackHeaderConfigComponentView willShowViewController:viewController animated:animated withConfig:config];
+    RNSScreenStackHeaderConfig *config = (RNSScreenStackHeaderConfig *)((RNSScreenView *)view).config;
```

Comment:
Could you debug why this code is added and the previous implementation is not used instead?

Reply:
I do not understand this completely, but paper implementation of `RNSScreenView` does not expose `config` property. In Fabric implementation this prop is updated when child component view is being mounted ([`mountChildComponentView` method](https://github.com/software-mansion/react-native-screens/blob/0c34961e4e87ee0563f11eb17b7fd675c0eb13dd/ios/RNSScreen.mm#L405-L412)) or unmounted ([`unmountChildComponentView` method](https://github.com/software-mansion/react-native-screens/blob/0c34961e4e87ee0563f11eb17b7fd675c0eb13dd/ios/RNSScreen.mm#L414-L420))